orion/controllers/announce/announceController.go

```diff
@@ -0,0 +1,107 @@
+package announce
+
+import (
+	"errors"
+	"net/http"
+	"regexp"
+	
+	"github.com/gin-gonic/gin"
+)
+
+const REGEX = `[A-Za-z]+`
+
+func GetAnnouncements(c *gin.Context) {
+	// Query Repo
+	announceList := GetAllAnnouncements()
+
+	// JSON Response
+	c.JSON(http.StatusOK, announceList)
+	return
+}
+
+func GetAnnouncement(c *gin.Context) {
+	// Incoming parameters
+	id := c.Param("id")
+
+	// Query Repo
+	announce, err := GetAnnouncementById(id)
+	if err != nil {
+		panic(err)
+	} else {
+		c.JSON(http.StatusOK, announce)
+	}
+	return
+}
+
+func CreateAnnouncement(c *gin.Context) {
+	// Incoming JSON
+	var announceJson Announce
+	c.BindJSON(&announceJson)
+
+	if err := CheckValidAnnouncement(announceJson); err != nil {
+		c.String(http.StatusBadRequest, err.Error())
+		return
+	}
+
+	// Query Repo (INSERT & SELECT)
+	err := InsertAnnouncement(announceJson)
+	if err != nil {
+		panic(err)
+	} else {
+		c.JSON(http.StatusOK, nil)
+	}
+	return
+}
+
+func UpdateAnnouncement(c *gin.Context) {
+	// Incoming JSON & Parameters
+	id := c.Param("id")
```

Comment:
`id` here is a parameter from the URL so it will first be a string.
I think we will have to parse this string and convert it into a uint.

Reply:
Yeah I decided to make id a string so I wouldn't have to parse and convert the string, but if it's better practice to make id type uint then I'll do that instead.